src/TurnerSoftware.SitemapTools/SitemapEntry.cs

```diff
@@ -32,5 +28,85 @@ public SitemapEntry()
 		{
 			Priority = 0.5;
 		}
+
+		#region Equality comparisons
+
+		public override int GetHashCode()
+		{
+			if (ReferenceEquals(this, null))
+				return default(Uri).GetHashCode();
+			return Location.GetHashCode();
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (ReferenceEquals(this, obj))
+				return true;
+		
+			{
+				if (ReferenceEquals(this, null))
+					return (obj is SitemapEntry other) && other.Location == null;
+			}
+
+			if (ReferenceEquals(obj, null))
+				return Location == null;
+
+			{
+				if (obj is SitemapEntry other)
+					return Location == other.Location;
+			}
+
+			{
+				if (obj is Uri other)
+					return Location == other;
+			}
+			return false;
+		}
+
+		public bool Equals(SitemapEntry other) => this == other;
+
+		public bool Equals(Uri other) => this == other;
+
+		public static bool operator ==(SitemapEntry x, SitemapEntry y)
+		{
+			if (ReferenceEquals(x, y))
+				return true;
+
+			{
+				if (ReferenceEquals(x, null))
+					return y.Location == null;
+			}
+
+			if (ReferenceEquals(y, null))
+				return x.Location == null;
+
+			return x.Location == y.Location;
+		}
+
+		public static bool operator !=(SitemapEntry x, SitemapEntry y) => !(x == y);
+
+		public static bool operator ==(SitemapEntry x, Uri y)
+		{
+			if (ReferenceEquals(x, y))
+				return true;
+
+			{
+				if (ReferenceEquals(x, null))
+					return false;
+			}
+
+			if (ReferenceEquals(y, null))
+				return x.Location == null;
+
+			return x.Location == y;
```

Comment:
As `SitemapEntry` and `Uri` are different types, I don't think the `ReferenceEquals` checks are necessary. As far as I can tell, it could be simplified to this:

```csharp
return (x == null && y == null) || x?.Location == y;
```

Reply:
I changed it.